aggregator/src/main/java/com/redhat/cloud/notifications/DailyEmailAggregationJob.java

```diff
@@ -20,38 +20,48 @@
 import java.util.logging.Logger;
 import java.util.stream.Collectors;
 
-import static com.redhat.cloud.notifications.models.EmailSubscriptionType.*;
+import static com.redhat.cloud.notifications.EmailSubscriptionType.*;
 import static java.time.ZoneOffset.UTC;
 import static java.time.temporal.ChronoUnit.*;
 
 @ApplicationScoped
 public class DailyEmailAggregationJob {
 
-    private static final Logger LOG = Logger.getLogger(DailyEmailAggregationJob.class.getName());
+    public static final String AGGREGATION_CHANNEL = "aggregation";
 
-    @Inject
-    EmailAggregationResources emailAggregationResources;
+    private static final Logger LOG = Logger.getLogger(DailyEmailAggregationJob.class.getName());
 
-    @Inject
-    ObjectMapper objectMapper;
+    private final EmailAggregationResources emailAggregationResources;
+    private final ObjectMapper objectMapper;
 
     @Inject
-    @Channel("aggregation")
+    @Channel(AGGREGATION_CHANNEL)
     Emitter<String> emitter;
 
-    public void processDailyEmail(Instant scheduledFireTime) {
-        if (isCronJobEnabled()) {
-            List<AggregationCommand> aggregationCommands = processAggregateEmails(scheduledFireTime);
-            List<CompletableFuture<Void>> futures = new ArrayList<>();
-            for (AggregationCommand aggregationCommand : aggregationCommands) {
-                try {
-                    final String payload = objectMapper.writeValueAsString(aggregationCommand);
-                    futures.add(emitter.send(payload).toCompletableFuture());
-                } catch (JsonProcessingException e) {
-                    LOG.warning("Could not transform AggregationCommand to JSON object.");
-                }
+    public DailyEmailAggregationJob(EmailAggregationResources emailAggregationResources, ObjectMapper objectMapper) {
+        this.emailAggregationResources = emailAggregationResources;
+        this.objectMapper = objectMapper;
+    }
+
+    public void processDailyEmail() {
+        if (!isCronJobEnabled()) {
+            return;
+        }
+        LocalDateTime now = LocalDateTime.now(UTC);
+        List<AggregationCommand> aggregationCommands = processAggregateEmails(now);
+
+        List<CompletableFuture<Void>> futures = new ArrayList<>();
+        for (AggregationCommand aggregationCommand : aggregationCommands) {
+            try {
+                final String payload = objectMapper.writeValueAsString(aggregationCommand);
+                futures.add(emitter.send(payload).toCompletableFuture());
+            } catch (JsonProcessingException e) {
+                LOG.warning("Could not transform AggregationCommand to JSON object.");
```

Comment:
> [This old comment](https://github.com/RedHatInsights/notifications-backend/pull/474#discussion_r701271991) (which is hidden by GitHub on my machine) requires your attention 😃

It's been burried by GitHub above so here's a copy of it:

This suggestion won't work out of the box because of JUL but we need to log the exception:
```suggestion
                LOG.warning("Could not transform AggregationCommand to JSON object.", e);
```